src/protocols/gamespy/protocols/one/protocol.rs

```diff
@@ -198,7 +199,10 @@ pub fn query_vars(
 /// Providing None to the timeout settings results in using the default values.
 /// (TimeoutSettings::[default](TimeoutSettings::default)).
 pub fn query(address: &SocketAddr, timeout_settings: Option<TimeoutSettings>) -> GDResult<Response> {
-    let mut server_vars = query_vars(address, timeout_settings)?;
+    let retry_count = TimeoutSettings::get_retries_or_default(&timeout_settings);
+    let mut server_vars = retry_on_timeout(retry_count, move || {
+        query_vars(address, timeout_settings.clone())
```

Comment:
The clone here is undesirable but needed because there is no struct to take ownership of it here.